pkg/apis/core/v1/defaults.go

```diff
@@ -429,3 +435,55 @@ func SetDefaults_HostPathVolumeSource(obj *v1.HostPathVolumeSource) {
 		obj.Type = &typeVol
 	}
 }
+
+// defaultPodResources applies default values for pod-level requests, when pod-level
```

Comment:
I'm having troiuble finding the issue link, but I believe we cannot default fields that are in the pod template, otherwise the workload controllers will detect that as a template update and do a rolling update. @thockin do you remember this issue?

Reply:
cc: @liggitt  as well

Reply:
Thanks @yujuhong! yeah which is why I didn't add the defaulting logic in Pod Spec even though the new field is getting added in PodSpec. marking this as resolved.